ts/Core/Series/DataLabel.ts

```diff
@@ -892,7 +892,12 @@ namespace DataLabel {
 
         // Off bottom
         off = (alignAttr.y || 0) + bBox.height - padding + verticalAxisShift;
-        if (off > chart.plotHeight) {
+        // Extend plot area's bottom to navigator when enabled. #21285
+        const isNavigatorEnabled = chart.navigator?.navigatorEnabled;
+        const threshold = isNavigatorEnabled ?
+            chart.yAxis[1].top + chart.yAxis[1].height : chart.plotHeight;
```

Comment:
1. Product-specific code should belong to that product. What I mean, is navigator-related if-els classes should not pollute Highcharts Core code. Extract this solution from here to Stock-related extensions. I think we don't have DataLabel for the Stock module, so the best place would be StockChart.ts file. Let's start there

2. Don't use `axis[index]` when trying to get naviagtor's xAxis. It won't work in Stock, when often we have multiple panes. You can get navigator's xAxis/yAxis this way: `chart.navigator.xAxis`.